app/routes/apiv1/course.js

```diff
@@ -0,0 +1,108 @@
+const express = require('express');
+
+const router = express.Router();
+const stripe = require('stripe');
+
+const config = require('./../../config');
+
+const Course = require('./../../models/Course');
+const Profile = require('./../../models/Profile');
+
+const createCourse = require('./../../util/course/createCourse');
+const studentEnrollWithCode = require('./../../util/course/studentEnrollWithCode');
+const overviewHelper = require('./../../util/course/overviewHelpers');
+
+const stripeHandler = stripe(config.stripe.secret_key);
+
+
+/** POST /api/course/create
+ *
+ * Creating a course, redirect to stripe.
+ */
+router.post('/create', (req, res) => {
+  const { name, description, plan } = req.body;
+
+  if (!name || name.trim().length === 0) {
+    return res.status(400).json({
+      success: false,
+      error: 'no-name',
+      message: 'Please provide a course name.',
+    });
+  }
+
+  if (!description) {
+    return res.status(400).json({
+      success: false,
+      error: 'no-description',
+      message: 'Please provide a course description.',
+    });
+  }
+
+  if (!plan || !(plan in config.plans)) {
+    return res.status(400).json({
+      success: false,
+      error: 'no-plan',
+      message: 'Please select a course plan.',
+    });
+  }
+
+  // Handle the case where the user that is creating a course is an
+  // administrator.
+  if (req.user.is_admin) {
```

Comment:
Validate the `coupon` received and compute a price relative to whether the user is administrator.